From: Mika Westerberg <mika.westerberg@linux.intel.com>
Date: Wed, 27 Feb 2013 17:25:15 +0200
Subject: gpio-ich: Fix ichx_gpio_check_available() return what callers expect
Patch-mainline: Merged by maintainer 2 Mar 2013 http://git.secretlab.ca/?p=linux.git;a=commit;h=e97f9b5277afeabb54892ebc6f68500098467ba1
Git-commit: e97f9b5277afeabb54892ebc6f68500098467ba1

ichx_gpio_check_available() returns either 0 or -ENXIO depending on whether
the given GPIO is available or not. However, callers of this function treat
the return value as boolean:

	...
	if (!ichx_gpio_check_available(gpio, nr))
		return -ENXIO;

which erroneusly fails when the GPIO is available and not vice versa.

Fix this by making the function return boolean as expected by the callers.

Signed-off-by: Mika Westerberg <mika.westerberg@linux.intel.com>
Signed-off-by: Grant Likely <grant.likely@secretlab.ca>
Acked-by: Jean Delvare <jdelvare@suse.de>
---
 drivers/gpio/gpio-ich.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/gpio/gpio-ich.c
+++ b/drivers/gpio/gpio-ich.c
@@ -128,9 +128,9 @@ static int ichx_read_bit(int reg, unsign
 	return data & (1 << bit) ? 1 : 0;
 }
 
-static int ichx_gpio_check_available(struct gpio_chip *gpio, unsigned nr)
+static bool ichx_gpio_check_available(struct gpio_chip *gpio, unsigned nr)
 {
-	return (ichx_priv.use_gpio & (1 << (nr / 32))) ? 0 : -ENXIO;
+	return ichx_priv.use_gpio & (1 << (nr / 32));
 }
 
 static int ichx_gpio_direction_input(struct gpio_chip *gpio, unsigned nr)
